### DISCUSSION TITLE
How to visualize the jooq expression tree?

### DISCUSSION THREAD

**Discussion by henry99923123** [jOOQ/jOOQ#15707] [Q&A] [answered]
Is there an easy way to visualize the jOOQ expression tree in the new Traverser API if you have a query like 

`context.select(Author.AUTHOR.LAST_NAME.as("lastname")).from(Author.AUTHOR);`

In a pretty tree-like structure in the console? Thank you.

**Comment by lukaseder** [comment, marked as answer]
Thank you very much for your message.

We currently do not have any out of the box textual representation of an expression tree other than the rendered SQL itself. There's pending work referenced from:

- https://github.com/jOOQ/jOOQ/issues/14756

Including:

- https://github.com/jOOQ/jOOQ/issues/855
- https://github.com/jOOQ/jOOQ/issues/642

Other formats than XML can also be supported. You can do this yourself by implementing your own `Traverser`, serialising the objects to your own custom data format. Let me know if you have any specific questions about that.
